### ISSUE TITLE
Introduce mxmake (one time) generated Makefile for the backend.

### ISSUE THREAD

**Issue by jensens** [collective/cookiecutter-plone-starter#133]
Meanwhile [mxmake](https://github.com/mxstack/mxmake) is very stable and rocks. We use it in many  Python and Zope projects. It generated makefile solves many problems the current PHONY file has, like using make but not its dependency management. To be honest, the current use of make has not many advantages over using a bash or python script. 

If mxmake is not able to fulfill a use case we have in here, we are happy to implement it.  Mxmake is currently still alpha, but this primary because @rnixx and I tend to give us still some flexibility to break things. When introduced here, we - in semver manner - would go beta/rc soon and before Plone 6.1 release latest a mxmake 1.0 final will be released.

mxmake generated Makefiles are even running on Windows... So I am pretty sure this will work for Plone too (and I am not talking about WSL here).

I propose for Plone 6.1 backend to switch to mxmake and document the process accordingly. This simplifies and speeds up the backend part, while also giving more flexibility. 

I would start this on a new plone-6.1 branch. Its not that much work, since mxmake is already kicking.

What do you think?

**Comment by gforcada** [comment]
Is this not made into a PLIP for buildout.coredev in purpose, or rather, as to first show some more upstream adoption for an upcoming PLIP? 😄

**Comment by jensens** [comment]
I think this is unrelated to buildout.coredev ?

**Comment by stevepiercy** [comment]
> Its generated makefile resolves many issues inherent in the current "PHONY" file here, particularly in terms of utilizing make without its dependency management. Frankly, the current utilization of make offers few advantages over employing a bash or Python script.

@jensens can you elaborate on this point? In `plone/documentation`, I've had trouble with `make` with symlinks, submodules, and checking for changes in dependencies, but I cannot recall specifics. My method to resolve issues was a sledgehammer, deleting my clone and starting over. Anyway, I would be interested in using `plone/documentation` as a test project.

**Comment by jensens** [comment]
@stevepiercy I do not think this change in here would touch any documentation setup related parts. Its about writing/ updating the docs about the usage of this cookiecutter.
